### ISSUE TITLE
2024-02-19 `sysbench-select-random-limits-medium-3cn` performance regression

### ISSUE THREAD

**Issue by lmatz** [risingwavelabs/risingwave#15140]
![SCR-20240220-dbw](https://github.com/risingwavelabs/risingwave/assets/5791930/52fbf763-2fc1-41c6-bfea-254e6b432833)

https://buildkite.com/risingwave-test/sysbench/builds/687#

**Comment by lmatz** [comment]
http://metabase.risingwave-cloud.xyz/question/5639-sysbench-select-random-limits-medium-3cn-sysbench-qps-transactions-s-history-thtb-1910?start_date=2024-01-08

somehow this test case fluctuates widely

![SCR-20240220-fs8](https://github.com/risingwavelabs/risingwave/assets/5791930/213c8987-1b4b-4f2c-80f0-594f5053f6ce)

**Comment by lmatz** [comment]
http://metabase.risingwave-cloud.xyz/question/5639-sysbench-select-random-limits-medium-3cn-sysbench-qps-transactions-s-history-thtb-1910?start_date=2024-01-08

02-15 is much better: https://grafana.test.risingwave-cloud.xyz/d/EpkBw5W4k/risingwave-dev-dashboard?from=1708033097000&orgId=1&to=1708033408000&var-datasource=Prometheus:+test-useast1-eks-a&var-namespace=sysbench-daily-test-20240215

02-19 is much worse: https://grafana.test.risingwave-cloud.xyz/d/EpkBw5W4k/risingwave-dev-dashboard?orgId=1&var-datasource=Prometheus:%20test-useast1-eks-a&from=1708378710000&to=1708379021000&var-namespace=sysbench-daily-test-20240219

## CPU

02-15:
![SCR-20240220-g0n](https://github.com/risingwavelabs/risingwave/assets/5791930/00ab3761-9b43-4c99-bf9d-dab1b89c01e4)

02-19:
![SCR-20240220-g0i](https://github.com/risingwavelabs/risingwave/assets/5791930/60e5bf98-f010-42b7-a9f4-5b18850c7196)

somehow 02-19's CPU usage is consistently higher.



## Hummock Read

02-15:
![SCR-20240220-g2v](https://github.com/risingwavelabs/risingwave/assets/5791930/e30016b4-8733-4653-90df-aa4e65b42593)


02-19:
![SCR-20240220-g2m](https://github.com/risingwavelabs/risingwave/assets/5791930/477ad465-c348-4668-814b-e9dc6376e825)


drastically different, which raises the question if there is any difference in compaction


## Compaction


02-15:
https://grafana.test.risingwave-cloud.xyz/d/EpkBw5W4k/risingwave-dev-dashboard?from=1708033097000&orgId=1&to=1708033408000&var-datasource=Prometheus:%20test-useast1-eks-a&var-namespace=sysbench-daily-test-20240215&editPanel=131
![SCR-20240220-g60](https://github.com/risingwavelabs/risingwave/assets/5791930/228af6a7-6f71-4c3f-8f64-5dcf6ed99978)


02-19:
https://grafana.test.risingwave-cloud.xyz/d/EpkBw5W4k/risingwave-dev-dashboard?orgId=1&var-datasource=Prometheus:%20test-useast1-eks-a&from=1708378710000&to=1708379021000&var-namespace=sysbench-daily-test-20240219&editPanel=131
![SCR-20240220-g63](https://github.com/risingwavelabs/risingwave/assets/5791930/043fcb31-c28b-467d-8797-0b756fb5335f)

**Comment by lmatz** [comment]
searched the issue history and the fluctuation happened a while ago: https://github.com/risingwavelabs/risingwave/issues/14022

@cyliu0  did the config change again? 
or shall we change it again? 
The CPU is right now not fully utilized I suppose, 
since `BENCH_TESTBED="medium-3cn"` and `medium` machines have 8 cpus while currently only 600% is used.

also cc: @chenzl25

**Comment by cyliu0** [comment]
No... we didn't change it since the above issue. It's still 64 threads. Which config would you like to change?
<img width="809" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/35356271/3259a96a-26d2-4a66-9707-7273b848993a">

**Comment by lmatz** [comment]
>  Which config would you like to change?

I am curious about the story behind why we changed the thread number in the previous issue: #14022

And also how was it determined that the problem was gone last time?

**Comment by cyliu0** [comment]
> > Which config would you like to change?
> 
> I am curious about the story behind why we changed the thread number in the previous issue: #14022

Originally from here https://risingwave-labs.slack.com/archives/C03GTKX3C8G/p1702015068556349
 
> And also how was it determined that the problem was gone last time?

+1 @chenzl25

**Comment by chenzl25** [comment]
Maybe we can increase the threads for this test? because its performance is better than before now and our resources are not fully utilized/

**Comment by lmatz** [comment]
> Maybe we can increase the threads number for this test? because its performance is better than before now and our resources are not fully utilized/

increase the `sys_threads` from 64 to an even larger number, e.g. 128 cc: @cyliu0 , could you run the new setting once, let's see if CPU can be fully utilized under such setting

**Comment by cyliu0** [comment]
The highest one `10667.46` is the 128 threads result https://buildkite.com/risingwave-test/sysbench/builds/689#018dc9f0-aa60-45d3-94c8-7848b15bc601

<img width="1165" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/35356271/60c9df85-2c06-4c16-9af3-7251374b16c7">

**Comment by cyliu0** [comment]
It's still not stable after we changed the threads to 128 in daily test. http://metabase.risingwave-cloud.xyz/dashboard/278-sysbench-rw-qps
<img width="757" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/35356271/2dd955f2-e6c7-4a60-8f48-221f2c68d58a">

**Comment by lmatz** [comment]
https://grafana.test.risingwave-cloud.xyz/d/EpkBw5W4k/risingwave-dev-dashboard?orgId=1&var-datasource=Prometheus:%20test-useast1-eks-a&from=1709501871000&to=1709502182000&var-namespace=sysbench-daily-test-20240303
![SCR-20240311-cx1](https://github.com/risingwavelabs/risingwave/assets/5791930/9994cbff-3850-4069-868d-1975cdc0ab21)


CPU usage seems to be stable, but still fluctuate a lot

http://metabase.risingwave-cloud.xyz/question/5639-sysbench-select-random-limits-medium-3cn-sysbench-qps-transactions-s-history-thtb-1910?start_date=2024-01-08

![SCR-20240311-cxj](https://github.com/risingwavelabs/risingwave/assets/5791930/04e6fd39-0b27-4403-9069-e29bbed11f26)
